### PR TITLE
Improve `asDrizzleTable()` types

### PR DESCRIPTION
**Pull request by delucis**
## Changes

- Improves the typing of the `asDrizzleTable()` utility

- Fixes a type error when passing the output of `defineTable()` to the utility and returns a more detailed type inferred from the columns of the passed table config.

## Testing

Tested manually in an integration

## Docs

n/a — bug fix

### CODE REVIEW

File: packages/db/src/runtime/index.ts [hunk 1]
```diff
@@ -8,8 +8,14 @@ import {
 	sqliteTable,
 	text,
 } from 'drizzle-orm/sqlite-core';
-import { type DBColumn, type DBTable } from '../core/types.js';
-import { type SerializedSQL, isSerializedSQL } from './types.js';
+import { tableSchema } from '../core/schemas.js';
```
Comment (bholmesdev) [inline packages/db/src/runtime/index.ts:11]: Ah ha, it looks like `tableSchema` would pull a core module into the runtime. We have a lint check against this. Perhaps we can move `asDrizzleTable` from `runtime/` to `core/` and still expose as a package export?

Comment (delucis) [reply packages/db/src/runtime/index.ts:11]: Took a quick look, but not sure if that would work? `asDrizzleTable` is used in the Vite plugin:

https://github.com/withastro/astro/blob/cc83367c4fd48ec8bf493d620ef5615780388420/packages/db/src/core/integration/vite-plugin-db.ts#L122

Comment (delucis) [reply packages/db/src/runtime/index.ts]: OK, tried a different approach, which seems to work. I think the `asDrizzleTable()` we expose to users needs to be different from the one we use internally, so in the latest commit I wrapped the internal version with the tweaks we made for use _only_ by integration authors.

**Review by bholmesdev — APPROVED** [review]
Glad we could addr`izz` the issue!

**Pull request merged**